src/assets/js/contact-form.js

```diff
@@ -13,15 +13,21 @@ export class ContactForm {
     this.prefillService();
 
     this.bindEvents();
+
+    this.enableForm();
   }
 
   bindEvents() {
     this.form.addEventListener("submit", event => {
-      event.preventDefault();
-      this.updateFormState("loading", "Your message is being sent...");
-
-      const formData = new FormData(this.form);
-      this.sendMessage(Object.fromEntries(formData.entries()));
+      if (this.form.reportValidity()) {
+        event.preventDefault();
+        this.updateFormState("loading", "Your message is being sent...");
+
+        const formData = new FormData(this.form);
+        this.sendMessage(Object.fromEntries(formData.entries()));
+      } else {
+        event.preventDefault();
+      }
```

Comment:
It really calls `preventDefault` always here so it can be hoisted but overall looks good
```suggestion
      event.preventDefault();
      if (this.form.reportValidity()) {
        this.updateFormState("loading", "Your message is being sent...");

        const formData = new FormData(this.form);
        this.sendMessage(Object.fromEntries(formData.entries()));
      }
```